extractor/docker/docker.go

```diff
@@ -271,40 +274,68 @@ func (d DockerExtractor) ExtractFromFile(ctx context.Context, r io.Reader, filen
 		if err != nil {
 			return nil, extractor.ErrCouldNotExtract
 		}
+		tarFiles[header.Name], err = ioutil.ReadAll(tr)
```

Comment:
I understand that you want to do this. But, if an image is too large, I think all content can't be stored in memory. What do you think?